wagtail/contrib/postgres_search/backend.py

```diff
@@ -143,14 +143,21 @@ def add_items_update_then_create(self, content_type_pk, objs, config):
                     object_id=object_id,
                     body_search=ids_and_objs[object_id]._search_vector,
                 ))
-        index_entries.bulk_create(to_be_created)
+        self.index_entries.bulk_create(to_be_created)
 
     def add_items(self, model, objs):
-        content_type_pk = get_content_types_pk(model)
+        content_type_pk = get_content_type_pk(model)
         config = self.backend.get_config()
         for obj in objs:
             obj._object_id = force_text(obj.pk)
             obj._body_ = self.prepare_body(obj)
+
+        # Removes index entries of an ancestor model in case the descendant
```

Comment:
I think this will be better implemented by storing the root content type (eg, ``Page``) alongside the leaf content type (eg ``HomePage``). Then make each index entry unique on the root content type and ID.

This would mean that if the same object was indexed twice with different leaf types, only a single instance would be in the index.

I think we should solve the issue of indexing the same object twice by adding a better default implementation of ``get_indexed_objects`` on the ``Indexed`` class.